src/Xamarin.Android.Build.Tasks/Tasks/ConvertCustomView.cs

```diff
@@ -79,24 +79,23 @@ public override bool RunTask ()
 					}
 				}
 			}
-			var output = new List<ITaskItem> ();
+			var output = new Dictionary<string, ITaskItem> ();
 			foreach (var file in processed) {
 				ITaskItem resdir = ResourceDirectories?.FirstOrDefault (x => file.StartsWith (x.ItemSpec)) ?? null;
-				if (resdir == null) {
+				if (output.ContainsKey (file))
```

Comment:
If `processed` is a `HashSet<string>`:

https://github.com/xamarin/xamarin-android/blob/802f0d3af0a17c952673127b12f6058218262edd/src/Xamarin.Android.Build.Tasks/Tasks/ConvertCustomView.cs#L35

Do we need this `ContainsKey` check?

Reply:
I think we do, because we want to skip the code underneath if we already added a file of the same name.

Reply:
I don't think `processed` can't have a duplicate if it is a `HashSet`.

If you do: `Add ("Foo")` and `Add ("Foo")` it will have one item.

Reply:
yes, but there are a number of lines of code where we are calculating stuff before we add. If we do have a duplicate, lets at least NOT do those calculations again.

Reply:
actually ignore me, I just realised what you were saying 🤦‍♂